Yosemite/Yosemite/Payments/CardReaderConnectionStateMachine.swift

```diff
@@ -126,32 +181,26 @@ final class CardReaderConnectionController {
     private var gatewayID: String? {
         didSet {
             didSetGatewayID()
-            analyticsTracker.setGatewayID(gatewayID: gatewayID)
+//            analyticsTracker.setGatewayID(gatewayID: gatewayID)
         }
     }
 
-    init(
-        forSiteID: Int64,
-        storageManager: StorageManagerType = ServiceLocator.storageManager,
-        stores: StoresManager = ServiceLocator.stores,
+    public init(
+        siteID: Int64,
+        storageManager: StorageManagerType,
+        stores: StoresManager,
         knownReaderProvider: CardReaderSettingsKnownReaderProvider,
-        alertsPresenter: CardPresentPaymentAlertsPresenting,
-        alertsProvider: BluetoothReaderConnnectionAlertsProviding,
-        configuration: CardPresentPaymentsConfiguration,
-        analyticsTracker: CardReaderConnectionAnalyticsTracker
+        configuration: CardPresentPaymentsConfiguration
     ) {
-        siteID = forSiteID
+        self.siteID = siteID
         self.storageManager = storageManager
         self.stores = stores
         state = .idle
         knownCardReaderProvider = knownReaderProvider
-        self.alertsPresenter = alertsPresenter
-        self.alertsProvider = alertsProvider
         foundReaders = []
         knownReaderID = nil
         skippedReaderIDs = []
```

Comment:
Since we're cleaning up the init here, maybe those should also go into some sort of "initial state" function?